src/pages/Home/index.js

```diff
@@ -230,15 +235,72 @@ const Home = () => {
     });
   };
 
-  const createZip = async () => {
-    const packName = packMeta.name || DEFAULT_PACK_META.name;
-    const packDesc =
-      (packMeta.description ? packMeta.description + ' | ' : '') +
-      DEFAULT_PACK_META.description;
-    const packFormat = packMeta.pack_format || DEFAULT_PACK_META.pack_format;
+  const bedrockFileBuilder = async (root, packFormat, packDesc, packName, resolution) => {
```

Comment:
I notice these are independent of the `Home` component, so should probably be in their own file... but I'll do that myself later :)

I do like the idea of 'builder' functions, though.

Reply:
Yeah they could do with modularising a bit better. If you have an idea how to do it I'll leave it with you. I thought about it a bit but couldn't make up my mind.

Reply:
I'm leaving this to you :)